packages/testing/src/events.cairo

```diff
@@ -142,11 +167,14 @@ fn is_emitted<T, impl TEvent: starknet::Event<T>, impl TDrop: Drop<T>>(
     while i != self.events.len() {
         let (from, event) = self.events.at(i);
 
-        if from == expected_emitted_by
-            && event.keys == @expected_keys
-            && event.data == @expected_data {
-            is_emitted = true;
-            break;
+        if from == expected_emitted_by {
+            let keys_match = verify_indexed_keys(expected_event, event.keys);
```

Comment:
Note that this approach is redundant with the current implementation, since it is testing the same thing, which is that the event keys in the passed event are the same received in the emitted one. This doesn't address the issue of mistakenly changing a key from an event in the code, since the test would still pass. The approach I have in mind is something like the implementation we used to have here:

https://github.com/OpenZeppelin/cairo-contracts/blob/364db5b1aecc1335d2e65db887291d19aa28937d/src/tests/account/test_account.cairo#L547